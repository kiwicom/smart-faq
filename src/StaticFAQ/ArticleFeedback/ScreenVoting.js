// @flow

import * as React from 'react';
import css from 'styled-jsx/css';
import { ThumbUp, ThumbDown } from '@kiwicom/orbit-components/lib/icons';

import screensList from './screensList';

type Props = {|
  changeScreen: (nextScreen: string) => void,
|};

const style = css`
  div.initial-screen {
    display: flex;
    align-items: center;
  }
  p.question {
    font-size: 14px;
    color: #46515e;
  }
  div.feedback-button {
    color: #00a991;
    display: flex;
    align-items: center;
    cursor: pointer;
  }
  div.feedback-button p {
    margin-left: 9px;
    font-size: 14px;
  }
  div.feedback-button.thumb-up {
    margin-left: 23px;
  }
  div.feedback-button.thumb-down {
    margin-left: 28px;
  }
`;

const ScreenVoting = (props: Props) => (
  <div className="initial-screen">
    <p className="question">Was this article helpful?</p>
    <div
      className="feedback-button thumb-up"
      onClick={() => props.changeScreen(screensList.THANK_YOU)}
    >
      <ThumbUp size="medium" customColor="#00a991" />
      <p>Yes</p>
    </div>
    <div
      className="feedback-button thumb-down"
      onClick={() => props.changeScreen(screensList.FEEDBACK)}
    >
      <ThumbDown size="medium" customColor="#00a991" />
      <p>No</p>
    </div>
    <style jsx>{style}</style>
  </div>
);

export default ScreenVoting;
