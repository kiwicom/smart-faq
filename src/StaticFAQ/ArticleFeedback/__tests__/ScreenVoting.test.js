// @flow

import * as React from 'react';
import { shallow } from 'enzyme';

import ScreenVoting from '../ScreenVoting';

const ch = (sc: string) => {}; //eslint-disable-line

describe('ScreenVoting', () => {
  const component = shallow(<ScreenVoting changeScreen={ch} />);
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
