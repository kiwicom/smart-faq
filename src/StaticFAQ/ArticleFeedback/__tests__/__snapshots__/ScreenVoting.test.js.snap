// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenVoting should match snapshot 1`] = `
<div
  className="jsx-1712138937 initial-screen"
>
  <p
    className="jsx-1712138937 question"
  >
    Was this article helpful?
  </p>
  <div
    className="jsx-1712138937 feedback-button thumb-up"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    <ThumbUp
      customColor="#00a991"
      size="medium"
    />
    <p
      className="jsx-1712138937"
    >
      Yes
    </p>
  </div>
  <div
    className="jsx-1712138937 feedback-button thumb-down"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    <ThumbDown
      customColor="#00a991"
      size="medium"
    />
    <p
      className="jsx-1712138937"
    >
      No
    </p>
  </div>
  <JSXStyle
    css="div.initial-screen.jsx-1712138937{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}p.question.jsx-1712138937{font-size:14px;color:#46515e;}div.feedback-button.jsx-1712138937{color:#00a991;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;cursor:pointer;}div.feedback-button.jsx-1712138937 p.jsx-1712138937{margin-left:9px;font-size:14px;}div.feedback-button.thumb-up.jsx-1712138937{margin-left:23px;}div.feedback-button.thumb-down.jsx-1712138937{margin-left:28px;}"
    styleId="1712138937"
  />
</div>
`;
